pepper/cli.py

```diff
@@ -607,6 +607,21 @@ def run(self):
             for exit_code, ret in self.poll_for_returns(api, load):
                 yield exit_code, json.dumps(ret, sort_keys=True, indent=4)
         else:
-            ret = self.low(api, load)
             exit_code = 0
+            # keep trying until all expected nodes return
+            total_time = 0
+            start_time = time.time()
+            while True:
+                total_time = time.time() - start_time
+                if total_time > self.options.timeout:
+                    exit_code = 1
+                    break
+                try:
+                    ret = self.low(api, load)
```

Comment:
but it is making the same api call over and over again, which I don't think is desirable, you could potentially try to run a highstate twice in a row.

Reply:
Yep. Nice catch.

Reply:
But I understood your concern. It depends on environment.